modules/host/contractmanager/sectorupdate.go

```diff
@@ -375,6 +371,111 @@ func (wal *writeAheadLog) managedRemoveSector(id sectorID) error {
 	return nil
 }
 
+// managedRemoveSectors appends changes to the WAL to remove multiple sectors.
+// Individual sector and storage folder errors are ignored.
+func (wal *writeAheadLog) managedRemoveSectors(sectors map[sectorID]uint64) error {
+	wal.mu.Lock()
+	wal.cm.sectorMu.Lock()
+
+	changes := make([]sectorUpdate, 0, len(sectors))
+	for id, count := range sectors {
+		var exists bool
+		location, exists := wal.cm.sectorLocations[id]
+		if !exists {
+			wal.cm.log.Printf("cannot find location for sector %v", id)
+			continue
+		}
+
+		sf, exists := wal.cm.storageFolders[location.storageFolder]
+		if !exists || atomic.LoadUint64(&sf.atomicUnavailable) == 1 {
+			wal.cm.log.Printf("storage folder index %v for sector %v not found", location.storageFolder, id)
+			continue
+		}
+
+		removed := count
+		if location.count < removed {
+			removed = location.count
+		}
+
+		// Inform the WAL of the sector update.
+		location.count -= removed
+		su := sectorUpdate{
+			ID:     id,
+			Count:  location.count,
+			Folder: location.storageFolder,
+			Index:  location.index,
+		}
+		changes = append(changes, su)
+
+		// Update the in-memory representation of the sector.
+		if location.count == 0 {
+			// Delete the sector and mark it as available.
+			delete(wal.cm.sectorLocations, id)
+			sf.availableSectors[id] = location.index
+		} else {
+			// Reduce the sector usage.
+			wal.cm.sectorLocations[id] = location
+		}
+	}
+
+	if len(changes) == 0 {
+		wal.mu.Unlock()
+		wal.cm.sectorMu.Unlock()
+		return nil
+	}
+
+	wal.appendChange(stateChange{
+		SectorUpdates: changes,
+	})
+
+	ch := wal.syncChan
+	wal.mu.Unlock()
+	wal.cm.sectorMu.Unlock()
+	<-ch
```

Comment:
this certainly raised an eyebrow... why do we need to grab the WAL's channel here?

Reply:
It's a holdover from `managedRemoveSector` to wait until after the WAL has synchronized to update the sector metadata and clear the usage.